py-polars/src/conversion/any_value.rs

```diff
@@ -202,36 +210,23 @@ pub(crate) fn py_object_to_any_value<'py>(
     }
 
     fn get_timedelta(ob: &Bound<'_, PyAny>, _strict: bool) -> PyResult<AnyValue<'static>> {
-        Python::with_gil(|py| {
-            let f = UTILS
-                .bind(py)
-                .getattr(intern!(py, "timedelta_to_int"))
-                .unwrap();
-            let py_int = f.call1((ob, intern!(py, "us"))).unwrap();
-
-            let av = if let Ok(v) = py_int.extract::<i64>() {
-                AnyValue::Duration(v, TimeUnit::Microseconds)
-            } else {
-                // This should be faster than calling `timedelta_to_int` again with `"ms"` input.
-                let v_us = py_int.extract::<i128>().unwrap();
-                let v = (v_us / 1000) as i64;
-                AnyValue::Duration(v, TimeUnit::Milliseconds)
-            };
-            Ok(av)
-        })
+        let timedelta = ob.extract::<TimeDelta>()?;
+        if let Some(micros) = timedelta.num_microseconds() {
```

Comment:
Tried nanoseconds here, it broke the tests, so stuck to micro/milliseconds only.